doc/devices.rst

```diff
@@ -104,25 +104,142 @@ Each problem is unique, so it can often be best to choose the default behaviour
+
+.. code-block:: python
+
+    from mpi4py import MPI
+    import pennylane as qml
+    dev = qml.device('lightning.gpu', wires=8, mpi=True)
+    @qml.qnode(dev)
+    def circuit_mpi():
+      qml.PauliX(wires=[0])
+      return qml.state()
+    local_state_vector = circuit_mpi()
+
+Currently, a ``lightning.gpu`` device with multi-node and multi-GPUs backend supports all the ``gate operations`` and ``observables`` that a ``lightning.gpu`` device with a single GPU/node backend supports.
+
+By default, each MPI process will return the overall simulation results, except for the ``qml.state()`` and ``qml.prob()`` methods. Each MPI process will only return the local simulation
+results for the ``qml.state()`` and ``qml.prob()`` methods to avoid buffer overflow. It will be the users' responsibity to ensure the safety of data collection for those two methods. Here are examples of collecting
+the local simulation results for ``qml.state()`` and ``qml.prob()`` methods:
+
+The workflow for collecting local state vector (using the ``qml.state()`` method) to ``rank 0`` is as follows:
+
 .. code-block:: python
 
     from mpi4py import MPI
     import pennylane as qml
-    dev = qml.device('lightning.gpu', wires=8, mpi=True, mpi_buf_size=1)
+    comm = MPI.COMM_WORLD
+    rank = comm.Get_rank() 
+    dev = qml.device('lightning.gpu', wires=8, mpi=True)
     @qml.qnode(dev)
     def circuit_mpi():
       qml.PauliX(wires=[0])
       return qml.state()
     local_state_vector = circuit_mpi()
+    #rank 0 will collect the local state vector
+    state_vector = comm.gather(local_state_vector, root=0)
+    if rank == 0:
+      print(state_vector)
+    
+The workflow for collecting local probability (using the ``qml.prob()`` method) to ``rank 0`` is as follows:
+
+.. code-block:: python
+    
+    from mpi4py import MPI
+    import pennylane as qml
+    import numpy as np
+
+    comm = MPI.COMM_WORLD
+    rank = comm.Get_rank()
+    dev = qml.device('lightning.gpu', wires=8, mpi=True)
+    prob_wires = [0, 1]
+
+    @qml.qnode(dev)
+    def mpi_circuit():
+      qml.Hadamard(wires=1)
+      return qml.probs(wires=prob_wires)
+
+    local_probs = mpi_circuit()
+ 
+    #For data collection across MPI processes.
+    recv_counts = comm.gather(len(local_probs),root=0)
+    if rank == 0:
+      probs = np.zeros(2**len(prob_wires))
+    else:
+      probs = None
+
+    comm.Gatherv(local_probs,[probs,recv_counts],root=0)
+    if rank == 0:
+      print(probs)
+
+Then the python script can be executed with the following command:
+
+.. code-block:: console
+    
+    $ mpirun -np 4 python yourscript.py
+
+Furthermore, users can optimize the performance of their applications by allocating appropriate amount of GPU memory for MPI operation with the ``mpi_buf_size`` keyword argument. To allocate ``n`` megabytes (MB) of GPU memory for MPI operations, initialize a ``lightning.gpu`` device with the ``mpi_buf_size=n`` keyword argument, as follows:
```

Comment:
Is this value now accepting megabytes? Is there an API specific call that sets the scaling on this? Also, what is the default we use again? I assume also we are in MB, and not MiB (mebibyte)? 

In other words, are we scaling the value by 10^6, or are we scaling it so the overall value after sending to the C++ layer is a power of two?

Reply:
Yes, this value is in MB (megabytes) now and we scale this value by 10^6 at the cpp layer here [.](https://github.com/PennyLaneAI/pennylane-lightning-gpu/blob/c1544a96ca7dc044623a3e2cc315fa3a9dad8ea7/pennylane_lightning_gpu/src/simulator/MPIWorker.hpp#L37).

Reply:
ohhh, it's in mebibytes thanks! Just updated it.